tokio-trace/tokio-trace-core/src/event.rs

```diff
@@ -0,0 +1,95 @@
+//! Events represent single points in time during the execution of a program.
+use std::fmt;
+use {
+    dispatcher,
+    field::{self, Value},
+    Metadata,
+};
+
+/// `Event`s represent single points in time where something occurred during the
+/// execution of a program.
+///
+/// An `Event` can be compared to a log record in unstructured logging, but with
+/// two key differences:
+/// - `Event`s exist _within the context of a [`Span`]_. Unlike log lines, they
+///   may be located within the trace tree, allowing visibility into the
+///   _temporal_ context in which the event occurred, as well as the source
+///   code location.
+/// - Like spans, `Event`s have structured key-value data known as _fields_,
+///   which may include textual message. In general, a majority of the data
+///   associated with an event should be in the event's fields rather than in
+///   the textual message, as the fields are more structed.
+///
+/// [`Span`]: ::span::Span
+#[derive(Debug)]
+pub struct Event<'a> {
+    fields: field::ValueSet<'a>,
+    metadata: &'a Metadata<'a>,
+    message: Option<(field::Field, fmt::Arguments<'a>)>,
```

Comment:
This is necessary due to limitations on the lifetime of the `fmt::Arguments` produced by the `format_args!` macro --- it _has_ to be owned by the `Event` itself, as far as I can tell.

I spent a long time trying to figure out a way around having to special-case this, but I don't think it's possible.

Reply:
This is no longer necessary as of 07ba2556e45b3c67c2beed08aeb07f8271001b05 (thanks Carl for pointers!)